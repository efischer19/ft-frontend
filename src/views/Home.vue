<template>
  <div id="nav">
    <SignIn/>
    <h1>Fischer Things</h1>
    <section v-for="(posts, topic) in privatePosts" :key="topic">
      <h2 class="privateTopic">{{ topic }}</h2>
      <ul>
        <li v-for="post in posts" :key="post.path">
          <router-link
            :to="`posts/_${post.path}`"
          >
            <div class="postLink">
              <p class="postTitle">{{ post.title }}</p>
              <p class="postDate">{{ getDisplayDate(post.date) }}</p>
            </div>
          </router-link>
        </li>
      </ul>
    </section>
    <section v-for="(posts, topic) in publicPosts" :key="topic">
      <h2 class="publicTopic">{{ topic }}</h2>
      <ul>
        <li v-for="post in posts" :key="post.path">
          <router-link
            :to="`posts/${post.path}`"
          >
            <div class="postLink">
              <p class="postTitle">{{ post.title }}</p>
              <p class="postDate">{{ getDisplayDate(post.date) }}</p>
            </div>
          </router-link>
        </li>
      </ul>
    </section>
  </div>
</template>

<script>
import axios from 'axios';

import SignIn from '@/components/SignIn.vue';

import { getData, setData } from '@/util/expiringSessionCache';

export default {
  name: 'home',
  components: {
    SignIn,
  },
  props: {
    postPaths: {
      default: () => {},
      type: Array,
    },
  },
  data() {
    if (!this.publicPosts) {
      const cacheData = getData(`${this.$route.path}_public`);
      if (cacheData) { this.publicPosts = JSON.parse(cacheData); }
      axios.get('/api/public/posts.json').then(({ data }) => {
        setData(`${this.$route.path}_public`, JSON.stringify(data));
        this.publicPosts = data;
      });
    }

    if (!this.privatePosts) {
      const cacheData = getData(`${this.$route.path}_private`);
      if (cacheData) { this.privatePosts = JSON.parseData(cacheData); }
      const authToken = getData('ft-auth-token');
      if (authToken) {
        axios.get(
          'https://nccu1znzcj.execute-api.us-east-2.amazonaws.com/Prod/posts',
          {
            headers: {
              'X-Api-Key': 'oZE5pkcS5H4PlndobzKdH9wmoxO9uLqa2tY5wtaH',
              Authorization: authToken,
            },
          },
        ).then(({ data }) => {
          setData(`${this.$route.path}_private`, JSON.stringify(data));
          this.privatePosts = data;
        });
      }
    }
    return {
      publicPosts: this.postPaths,
      privatePosts: this.postPaths,
    };
  },
  methods: {
    getDisplayDate: (date) => {
      const options = { day: 'numeric', month: 'short' };
      return new Date(date).toLocaleDateString('en-US', options);
    },
  },
};
</script>

<style>
#nav h1 {
  margin: auto;
  width: 100%;
  margin-top: 20px;
  font-size: 3em;
  text-align: center;
  color: #404040;
}

#nav section {
  border: 5px solid #404040;
  border-radius: 20px;
  border-bottom-right-radius: 2px;
  border-bottom-left-radius: 2px;
  border-bottom-width: 5px;
  margin-top: 20px;
  background-color: #404040;
}

#nav h2 {
  color: #E6E6E6;
  margin: 10px;
}

#nav ul {
  list-style-type: none;
  padding: 0;
  background-color: #E6E6E6;
  margin-bottom: 0;
  padding-bottom: 1px;
}

#nav li {
  border-top: 2px solid darkgray;
}

.postLink {
  margin: 10px;
  overflow: hidden;
}

.postLink .postTitle {
  float: left;
}

.postLink .postDate {
  float: right;
  color: #404040;
}
</style>
