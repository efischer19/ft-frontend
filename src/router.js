import Vue from 'vue';
import Router from 'vue-router';
import Home from './views/Home.vue';
import About from './views/About.vue';
import Blog from './views/Blog.vue';

Vue.use(Router);

// just for local development
var localPostData = require('./mc_data.json');

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home,
    },
    {
      path: '/about',
      name: 'about',
      component: About,
    },
    {
      path: '/blog',
      name: 'blog',
      component: Blog,
      props: {
        baseImgDir: "mc/",
        postContent: localPostData,
      },
    },
  ],
});
