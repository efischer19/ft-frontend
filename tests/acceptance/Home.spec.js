import { createRenderer } from 'vue-server-renderer';
import Vue from 'vue';
import mockAxios from 'jest-mock-axios';
import Home from '@/views/Home.vue';
import router from '@/router';

const mockResponseObj = require('./posts.json');

describe('Home', () => {
  it('should match snapshot', () => {
    const Constructor = Vue.extend(Home);
    const vm = new Constructor({
      router,
      propsData: {
        id: 'mc',
      },
    }).$mount();

    expect(mockAxios.get).toHaveBeenCalledWith('/api/posts2.json');
    mockAxios.mockResponse({ data: mockResponseObj });

    const renderer = createRenderer();
    Vue.nextTick(() => {
      renderer.renderToString(
        vm,
        (err, str) => {
          if (err) throw new Error(err);
          expect(str).toMatchSnapshot();
        },
      );
    });
  });
});
