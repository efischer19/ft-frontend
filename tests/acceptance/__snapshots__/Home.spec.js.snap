// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div id="home">
  <nav>
    <div id="signin-header">
      <p>You are not signed in</p>
      <a href="https://signin.fischerthings.com/login?response_type=token&amp;client_id=6a8t6a9st8fqkbt81n9mtb2aag&amp;redirect_uri=https://fischerthings.com/login_success.html">Sign In</a>
    </div>
  </nav>
  <header>
    <h1>Fischer Things</h1>
  </header>
  <main>
    <section>
      <h2 class="topic">Cooking</h2>
      <ul>
        <li>
          <a href="#/posts/mc">
            <div class="postLink">
              <p class="postTitle">Mac and Cheese</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
        <li>
          <a href="#/posts/mc2">
            <div class="postLink">
              <p class="postTitle">Not Mac and Cheese</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
        <li>
          <a href="#/posts/mc3">
            <div class="postLink">
              <p class="postTitle">Just Cheese</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h2 class="topic">Baby Pictures</h2>
      <ul>
        <li>
          <a href="#/posts/mc4">
            <div class="postLink">
              <p class="postTitle">Baby Mac</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
        <li>
          <a href="#/posts/mc5">
            <div class="postLink">
              <p class="postTitle">Baby Cheese</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
        <li>
          <a href="#/posts/mc6">
            <div class="postLink">
              <p class="postTitle">Smol Noodlins</p>
              <p class="postDate">Jul 24</p>
            </div>
          </a>
        </li>
      </ul>
    </section>
  </main>
  <footer>
    <p>Â© Eric Fischer 2018</p>
    <a href="#/posts/faq">FAQ</a>
  </footer>
</div>
`;
